source/OptimaJet.DWKit.StarterApplication/Services/GroupService.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal;
+using JsonApiDotNetCore.Services;
+using Microsoft.Extensions.Logging;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class GroupService : EntityResourceService<Group>
+    {
+        public IOrganizationContext OrganizationContext { get; private set; }
+        public IJsonApiContext JsonApiContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public UserRepository UserRepository { get; }
+        public GroupRepository GroupRepository { get; }
+
+        public GroupService(
+            IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
+            UserRepository userRepository,
+            IEntityRepository<Group> groupRepository,
+            ILoggerFactory loggerFactory) : base(jsonApiContext, groupRepository, loggerFactory)
+        {
+            OrganizationContext = organizationContext;
+            JsonApiContext = jsonApiContext;
+            CurrentUserContext = currentUserContext;
+            UserRepository = userRepository;
+            GroupRepository = (GroupRepository)groupRepository;
+        }
+
+
+        public override async Task<IEnumerable<Group>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<Group>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
+                query.Filters.Add(new JsonApiDotNetCore.Internal.Query.FilterQuery("organization-header", value, "="));
+                return await base.GetAsync();
+            }
+        }
+        public override async Task<Group> GetAsync(int id)
+        {
+            var groups = await GetAsync();
+            return groups.SingleOrDefault(g => g.Id == id);
+        }
+        public override async Task<Group> UpdateAsync(int id, Group resource)
+        {
+            var updateForm = new Forms.Groups.UpdateForm(UserRepository,
```

Comment:
is there a particular reason for using the (nearly) fully qualified name?

Reply:
This is because until yesterday this was in the BackendServicesExtension and forms with the same name were being created for Projects and Groups in the same routine.  When I cut them out of there and changed it to a get new, I didn't think about the fact that there was no longer a need for the longer signature. Fixed